x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,390 @@
+
+      const defaultDataViewId = await deps.data.dataViews.getDefaultId();
+      const validDataViewId: string =
+        isDefined(state.dataViewId) && state.dataViewId !== ''
+          ? state.dataViewId
+          : defaultDataViewId ?? '';
+      let initialDataView: DataView[] | undefined;
+      try {
+        const dataView = isESQLQuery(state.query)
+          ? await getESQLAdHocDataview(
+              getIndexPatternFromESQLQuery(state.query.esql),
+              deps.data.dataViews
+            )
+          : await deps.data.dataViews.get(validDataViewId);
+        initialDataView = [dataView];
+      } catch (error) {
+        // Only need to publish blocking error if viewtype is data view, and no data view found
+        if (state.viewType === FieldStatsInitializerViewType.DATA_VIEW) {
+          onError(error);
+        }
+      }
+
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(initialDataView);
+
+      const subscriptions = new Subscription();
+      if (fieldStatsControlsApi.dataViewId$) {
+        subscriptions.add(
+          fieldStatsControlsApi.dataViewId$
+            .pipe(
+              skip(1),
+              skipWhile((dataViewId) => !dataViewId && !defaultDataViewId),
+              switchMap((dataViewId) => {
+                try {
+                  return deps.data.dataViews.get(dataViewId ?? defaultDataViewId);
+                } catch (error) {
+                  return of(undefined);
+                }
+              })
+            )
+            .subscribe((nextSelectedDataView) => {
+              if (nextSelectedDataView) {
+                dataViews$.next([nextSelectedDataView]);
+              }
+            })
+        );
+      }
+
+      const { toasts } = deps.notifications;
+
+      const api = buildApi(
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...fieldStatsControlsApi,
+          // PublishesDataLoading
+          dataLoading,
+          // PublishesBlockingError
+          blockingError,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.dataVisualizer.fieldStats.typeDisplayName', {
+              defaultMessage: 'field statistics',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableFieldStatsUserInput } = await import(
+                './resolve_field_stats_embeddable_input'
+              );
+              const chartState = serializeFieldStatsChartState();
+              const nextUpdate = await resolveEmbeddableFieldStatsUserInput(
+                coreStart,
+                pluginStart,
+                parentApi,
+                uuid,
+                false,
+                chartState,
+                fieldStatsControlsApi
+              );
+
+              fieldStatsControlsApi.updateUserInput(nextUpdate);
+            } catch (e) {
+              return Promise.reject(e);
```

Comment:
`Promise.reject(e)` will cause `onEdit` to throw. [EditPanelAction.execute](https://github.com/elastic/kibana/blob/main/src/plugins/presentation_panel/public/panel_actions/edit_panel_action/edit_panel_action.ts#L87) does not have any try/catch around `onEdit` calls, resulting in unhandled error. How about displaying a warning toast when editing fails instead of throwing?

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/1cad8784ad2ecc05fa10040827a7a2fc0d434c7b

Reply:
I added an error toast instead of warning toast, since that means the update fails for some reason and we probably would like to see the error trace. Let me know if that's not the standard and I can update it to be a warning toast.